src/corpora/auto.py

```diff
@@ -58,103 +84,98 @@ def get_auto_dataset(
         assert len(dataset) > 0, "You can't set ignore_train = True when there is only train data"
 
     # First, Normalize Text if Necessary. Tokenization Strategies are in detokenization.py.
-    dataset = auto_detokenize(dataset_id, dataset, paths["preprocessed"], preprocessing_num_proc)
+    dataset = auto_detokenize(dataset_id, dataset, paths["preprocessed"], preprocessing_num_proc, streaming)
 
     # Second, run straight-up tokenization
-    def tokenize(examples: Dict[str, List[str]]) -> BatchEncoding:
-        return tokenizer(examples["text"])
-
-    overwatch.info(f"Tokenizing Dataset via Multiprocessing with `{preprocessing_num_proc}` threads...")
+    if streaming:
+        tokenized_dataset = dataset.map(
+            _TokenizerFun(tokenizer),
+            batched=True,
+            remove_columns=list(next(iter(next(iter(dataset.values())))).keys()),
```

Comment:
This line is terrifying -- what's the double `next(iter(` doing?

Reply:
yeah sorry, i need to refactor into a method, i just didn't b/c draft. it's just getting the keys of the the first example of the first split and saying that's the column names. It's identical in intention to the line in the other branch with column_names, but IterableDatasets don't expose column_names by default (which I understand, but it's still annoying)

Reply:
Got it - makes sense.